agents/src/pipeline/pipeline_agent.ts

```diff
@@ -693,31 +693,31 @@ export class VoicePipelineAgent extends (EventEmitter as new () => TypedEmitter<
         this.emit(VPAEvent.FUNCTION_CALLS_FINISHED, calledFuncs);
         if (!newFunctionCalls) break;
       }
+    }
 
-      if (handle.addToChatCtx && (!userQuestion || handle.userCommitted)) {
-        this.chatCtx.messages.push(...extraToolsMessages);
-        if (interrupted) {
-          collectedText + '…';
-        }
-
-        const msg = ChatMessage.create({ text: collectedText, role: ChatRole.ASSISTANT });
-        this.chatCtx.messages.push(msg);
+    if (handle.addToChatCtx && (!userQuestion || handle.userCommitted)) {
```

Comment:
This was nested in an if when I think it wasn't supposed to be so AGENT_SPEECH_INTERRUPTED/COMMITTED were never firing